magento/pwa-studio
# Use ApolloClient for MagentoRoute

## Description

`MagentoRoute` and its talon execute a GQL query and fetch a RootComponent on route change. The code for doing these things is old, running queries and caching the results in local storage manually, which makes it difficult to clear cached results when admins change routes in the backend.

This PR attempts to update these calls to our standards.

## Related Issue

[PWA-821](https://jira.corp.magento.com/browse/PWA-821)

## Acceptance

### Verification Stakeholders

- @dpatil-magento 
- @sirugh 

### Specification

### Verification Steps

1. Clear site data.
2. Click links to navigate from the home page to a category, then to a product page.
3. Add a product to your cart, then click a link to the cart page.
4. Click the link in your cart items (product name) to go back to the product page.
5. Click the link in the breadcrumbs (category name) to go back to the category page.

- Navigation should be quick.
- There should be only one `ResolveURL` call in the Network tab for each navigation.
- Revisiting a page (steps 4 and 5) should be instant, as a cache hit.
- There should be no regressions when changing store code, whether in the URL or not.

## Screenshots / Screen Captures (if appropriate)

## Checklist

* I have added tests to cover my changes, if necessary.
* I have added translations for new strings, if necessary.
* I have updated the documentation accordingly, if necessary.
